core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -176,8 +176,10 @@ public NativeImageBuildItem build(NativeConfig nativeConfig, NativeImageSourceJa
                 //todo: this should be specific build items
                 if (prop.getKey().equals("quarkus.ssl.native") && prop.getValue() != null) {
                     enableSslNative = Boolean.parseBoolean(prop.getValue());
-                } else if (prop.getKey().equals("quarkus.jni.enable") && prop.getValue() != null) {
-                    nativeConfig.enableJni |= Boolean.parseBoolean(prop.getValue());
+                } else if (prop.getKey().equals("quarkus.jni.enable") && prop.getValue().equals("false")) {
+                    log.warn("Your application is setting the deprecated 'quarkus.jni.enable' configuration key to false."
+                            + " Please consider removing this configuration key as it is ignored (JNI is always enabled) and it"
+                            + " will be removed in a future Quarkus version.");
```

Comment:
Just to be sure: have you tested that you don't have several identical messages when setting `quarkus.jni.enable=false`?

Reply:
I did it with #6306 but I'll restest it again with this PR just be sure.

Reply:
I couldn't trigger this warning using `quarkus.jni.enable=false`, the only one being displayed is the one from `JniProcessor`. So, there's no double warning but I don't know if we need to keep this one.

Reply:
Unless I missed something, this warning is never displayed with the current Quarkus codebase. I suppose this is an old property that is no longer used by the core extensions, but it can still be used by 3rd parties extensions.

On the other hand, the warning from `JniProcessor` is displayed when `quarkus.jni.enable` is set to `false`.

I don't think we should worry about duplicate warnings here since it's very unlikely to happen.